lib/raygun.ts

```diff
@@ -0,0 +1,242 @@
+/*jshint unused:vars */
+
+/*
+ * raygun
+ * https://github.com/MindscapeHQ/raygun4node
+ *
+ * Copyright (c) 2015 MindscapeHQ
+ * Licensed under the MIT license.
+ */
+
+"use strict";
+
+import type {
+  RawUserData,
+  OfflineStorageOptions,
+  Tag,
+  CustomData,
+  RequestParams,
+  Message,
+} from "./types";
+import type { Request, Response, NextFunction } from "express";
+import * as raygunTransport from "./raygun.transport";
+import { RaygunMessageBuilder } from "./raygun.messageBuilder";
+import { OfflineStorage } from "./raygun.offline";
+
+type SendCB = (error: Error | null, items: string[] | undefined) => void;
+
+type Hook<T> = (
+  message: Message,
+  exception: Error | string,
+  customData: CustomData,
+  request?: RequestParams,
+  tags?: Tag[]
+) => T;
+
+type RaygunOptions = {
+  apiKey: string;
+  filters?: string[];
+  host?: string;
+  port?: number;
+  useSSL?: boolean;
+  onBeforeSend?: Hook<Message>;
+  offlineStorage?: OfflineStorage;
+  offlineStorageOptions?: OfflineStorageOptions;
+  isOffline?: boolean;
+  groupingKey?: Hook<string>;
+  tags?: Tag[];
+  useHumanStringForObject?: boolean;
+  reportColumnNumbers?: boolean;
+  innerErrorFieldName?: string;
+};
+
+class Raygun {
+  _apiKey: string | undefined;
+  _filters: string[] = [];
+  _user: RawUserData | undefined;
+  _version: string = "";
+  _host: string | undefined;
+  _port: number | undefined;
+  _useSSL: boolean | undefined;
+  _onBeforeSend: Hook<Message> | undefined;
+  _offlineStorage: OfflineStorage | undefined;
+  _isOffline: boolean | undefined;
+  _offlineStorageOptions: OfflineStorageOptions | undefined;
+  _groupingKey: Hook<string> | undefined;
+  _tags: Tag[] | undefined;
+  _useHumanStringForObject: boolean | undefined;
+  _reportColumnNumbers: boolean | undefined;
+  _innerErrorFieldName: string | undefined;
+
+  init(options: RaygunOptions) {
+    this._apiKey = options.apiKey;
+    this._filters = options.filters || [];
+    this._host = options.host;
+    this._port = options.port;
+    this._useSSL = options.useSSL !== false;
+    this._onBeforeSend = options.onBeforeSend;
+    this._offlineStorage = options.offlineStorage || new OfflineStorage();
+    this._offlineStorageOptions = options.offlineStorageOptions;
+    this._isOffline = options.isOffline;
+    this._groupingKey = options.groupingKey;
+    this._tags = options.tags;
+    this._useHumanStringForObject =
+      options.useHumanStringForObject === undefined
+        ? true
+        : options.useHumanStringForObject;
+    this._reportColumnNumbers = options.reportColumnNumbers;
+    this._innerErrorFieldName = options.innerErrorFieldName; // VError function to retrieve inner error;
```

Comment:
Just to confirm, we no longer need the `|| 'cause'` assignment here?

Reply:
Ah I had moved this because at one point the message builder required `innerErrorFieldName` to be a string, and since this class uses the `init` pattern `this._innerErrorFieldName` might be possibly `undefined`. The message builder handles not being passed an `innerErrorFieldName` now, so I've undone this change.